### PR TITLE
Adding Global Exception Handler and more methods in LoginController

### PR DESCRIPTION
**Pull request by minja1306**

### CODE REVIEW

File: src/main/java/eu/execom/pomodoroTeam/controllers/GlobalExceptionHandler.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
+package eu.execom.pomodoroTeam.controllers;
+
+import org.jboss.logging.Logger;
```
Comment (stanislavgrujic) [inline src/main/java/eu/execom/pomodoroTeam/controllers/GlobalExceptionHandler.java]: This is not a good logging class. You should use SLF4J, and you can achieve it just by adding a Lombok annotation `@Slf4j` on your class.

File: src/main/java/eu/execom/pomodoroTeam/controllers/GlobalExceptionHandler.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+import javax.persistence.EntityNotFoundException;
+
+@RestControllerAdvice
+public class GlobalExceptionHandler extends ResponseEntityExceptionHandler {
+    private static Logger log = Logger.getLogger(GlobalExceptionHandler.class);
+
+    @ExceptionHandler(EntityNotFoundException.class)
+    public ResponseEntity<Object> handleEntityNotFoundException(EntityNotFoundException e, WebRequest request) {
+        log.info("Entity not found!");
```
Comment (stanislavgrujic) [inline src/main/java/eu/execom/pomodoroTeam/controllers/GlobalExceptionHandler.java]: This log line does not give any information about which entity was not found. Could you extract an id from the exception, of from the request?

File: src/main/java/eu/execom/pomodoroTeam/controllers/GlobalExceptionHandler.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+        return new ResponseEntity<>(HttpStatus.NOT_FOUND);
+    }
+
+    @ExceptionHandler(Exception.class)
+    public ResponseEntity<Object> handleInternalServerErrorException(Exception e, WebRequest request) {
+        log.info("Internal server error!");
```
Comment (stanislavgrujic) [inline src/main/java/eu/execom/pomodoroTeam/controllers/GlobalExceptionHandler.java]: This does not give any information what happened to cause the error. Rather log the exception itself, e.g.:
```
log.error("There was an exception: ", e)
```

File: src/main/java/eu/execom/pomodoroTeam/controllers/GlobalExceptionHandler.java (hunk 2 above)
Comment (stanislavgrujic) [reply src/main/java/eu/execom/pomodoroTeam/controllers/GlobalExceptionHandler.java]: Also, use `error` level to log exceptions.

File: src/main/java/eu/execom/pomodoroTeam/controllers/GlobalExceptionHandler.java [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+        return new ResponseEntity<>(HttpStatus.INTERNAL_SERVER_ERROR);
+    }
+
+    private void logException(Exception e) {
+        log.info("Exception" + e + "occurred!");
```
Comment (stanislavgrujic) [inline src/main/java/eu/execom/pomodoroTeam/controllers/GlobalExceptionHandler.java]: This method is not used anywhere - either remove it, or use it. Also, to log exceptions, use error level, and logger (from SLF4j) provides a method to also print the stacktrace. E.g., 
```
log.error("Exception occured: ", e);
```

File: src/main/java/eu/execom/pomodoroTeam/controllers/LoginController.java [hunk 5]
```diff
@@ -34,22 +43,48 @@ public Principal user(Principal principal) {
         return principal;
     }
 
-    @GetMapping("/user/getAll")
-    public Iterable<UserEntity> getAllUsers() {
-        return userRepository.findAll();
-
+    @GetMapping("/getAll")
+    public ResponseEntity<List<UserEntity>> getAllUsers() {
+        return new ResponseEntity<>(userRepository.findAll(), HttpStatus.OK);
```
Comment (stanislavgrujic) [inline src/main/java/eu/execom/pomodoroTeam/controllers/LoginController.java]: You should not chain method calls, but rather assign users to a variable, and then pass it to a constructor.

File: src/main/java/eu/execom/pomodoroTeam/controllers/LoginController.java [hunk 6]
```diff
@@ -34,22 +43,48 @@ public Principal user(Principal principal) {
...
+    @Transactional(readOnly = true)
+    public ResponseEntity<?> getUserById(@PathVariable Long id) {
+        UserEntity user = userRepository.getOne(id);
         return new ResponseEntity<>(user, HttpStatus.OK);
     }
-}
\ No newline at end of file
+
+    @PutMapping(value = "/{id}")
+    public ResponseEntity<?> updateUser(@PathVariable Long id, @RequestBody UserDto userDto) {
```
Comment (stanislavgrujic) [inline src/main/java/eu/execom/pomodoroTeam/controllers/LoginController.java]: Please return concrete type whenever possible (e.g., `ResponseEntity<UserEntity>`)

File: src/main/java/eu/execom/pomodoroTeam/controllers/LoginController.java (hunk 5 above)
Comment (minja1306) [reply src/main/java/eu/execom/pomodoroTeam/controllers/LoginController.java]: I am not sure how to handle this. If i need return list, i have to put list as a ResponseEntity type.

File: src/main/java/eu/execom/pomodoroTeam/controllers/LoginController.java [hunk 7]
```diff
@@ -34,22 +45,50 @@ public Principal user(Principal principal) {
         return principal;
     }
 
-    @GetMapping("/user/getAll")
-    public Iterable<UserEntity> getAllUsers() {
-        return userRepository.findAll();
+    @GetMapping("/getAll")
 
+    public ResponseEntity<List<UserEntity>> getAllUsers() {
```
Comment (stanislavgrujic) [inline src/main/java/eu/execom/pomodoroTeam/controllers/LoginController.java]: Please avoid having floating annotations, there should not be any spacing between annotations and classes, fields or methods to which they are related.

File: src/main/java/eu/execom/pomodoroTeam/controllers/LoginController.java [hunk 8]
```diff
@@ -34,22 +45,50 @@ public Principal user(Principal principal) {
...
     }
 
-    @GetMapping("/user/getAll")
-    public Iterable<UserEntity> getAllUsers() {
-        return userRepository.findAll();
+    @GetMapping("/getAll")
 
+    public ResponseEntity<List<UserEntity>> getAllUsers() {
+        List<UserEntity> users = userRepository.findAll();
+        return new ResponseEntity<>(users, HttpStatus.OK);
```
Comment (stanislavgrujic) [inline src/main/java/eu/execom/pomodoroTeam/controllers/LoginController.java]: Please use `UserDto` instead of `UserEntity`, as a return value.

**Pull request merged**